### PR TITLE
GdbServer: Minor work

### PR DESCRIPTION
**Pull request by Sonicadvance1**
Unlike in #3592, this does some initial work towards making my life easier for improving the gdbserver.

Some very minor behaviour changes but pretty much just moving some code around and documenting the command packets so I can quickly search around in vim.

### CODE REVIEW

**Comment by neobrain** [comment]
Could you apply Draft status to this PR if you're still planning to add further patches?

**Comment by Sonicadvance1** [comment]
> Could you apply Draft status to this PR if you're still planning to add further patches?

I'm moving on to a different branch now

File: Source/Tools/LinuxEmulation/GdbServer/Info.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * @param Bit Which bit of EFLAG to query
+ */
+const std::string_view& GetFlagName(unsigned Bit);
+
+/**
+ * @brief Returns the textual name of a GPR register
+ *
+ * @param Reg Index of the register to fetch
+ */
+std::string_view const GetGRegName(unsigned Reg);
```
Comment (lioncash) [inline Source/Tools/LinuxEmulation/GdbServer/Info.h:32]:
```suggestion
std::string_view GetGRegName(unsigned Reg);
```

File: Source/Tools/LinuxEmulation/GdbServer/Info.cpp [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+#include <FEXCore/fextl/sstream.h>
+#include <FEXCore/Utils/CompilerDefs.h>
+#include <FEXCore/Utils/FileLoading.h>
+#include <FEXCore/Utils/LogManager.h>
+
+#include <array>
+#include <string_view>
+
+namespace FEX::GDB::Info {
+constexpr std::array<std::string_view const, 22> FlagNames = {
```
Comment (lioncash) [inline Source/Tools/LinuxEmulation/GdbServer/Info.cpp:25]:
```suggestion
constexpr std::array<std::string_view, 22> FlagNames = {
```

File: Source/Tools/LinuxEmulation/LinuxSyscalls/GdbServer.cpp [hunk 3]
```diff
@@ -648,11 +396,100 @@ void GdbServer::buildLibraryMap() {
   LibraryMapChanged = false;
 }
 
+// Binary data transfer handlers
+
+GdbServer::HandledPacketType GdbServer::XferCommandExecFile(const fextl::string& annex, int offset, int length) {
+  int annex_pid;
+  if (annex == "") {
```
Comment (lioncash) [inline Source/Tools/LinuxEmulation/LinuxSyscalls/GdbServer.cpp:403]:
```suggestion
  if (annex.empty()) {
```

File: Source/Tools/LinuxEmulation/LinuxSyscalls/GdbServer.cpp [hunk 4]
```diff
@@ -814,7 +590,106 @@ GdbServer::HandledPacketType GdbServer::handleProgramOffsets() {
...
+GdbServer::HandledPacketType GdbServer::CommandThreadOp(const fextl::string& packet) {
+  const auto match = [&](const char* str) -> bool {
+    return packet.rfind(str, 0) == 0;
+  };
+
+  if (match("Hc")) {
+    // Sets thread to this ID for stepping
+    // This is deprecated and vCont should be used instead
+    auto ss = fextl::istringstream(packet);
+    ss.seekg(fextl::string("Hc").size());
```
Comment (lioncash) [inline Source/Tools/LinuxEmulation/LinuxSyscalls/GdbServer.cpp:665]: should this be a string_view like down below (or use std::strlen directly)?

**Review by lioncash — COMMENTED** [review]
few minor nits for a first pass

**Pull request merged**